server/app.js

```diff
@@ -0,0 +1,109 @@
+const express = require('express');
+const cors = require('cors');
+const { Router } = require('express');
+const fetch = require('node-fetch');
+
+const TOKEN =
+	'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI2MzM3YjlhNS05ZTc5LTRlZTEtYWRmOC05ZDE5YmE1YzJjZGQiLCJwbGF0Zm9ybSI6InN1YnNjcmlwdGlvbnMiLCJpYXQiOjE2Nzg3ODQ2OTMsImV4cCI6MTY3OTY4NDY5M30.jof6BupoFmpBmnxkfNHRJZJm7V7R8V8S0_e60ROgo34';
+
+const app = express();
+const router = Router();
+
+app.use(cors());
+app.use(express.urlencoded({ extended: true }));
+
+router.get('/', async (req, res, next) => {
+	try {
+		const response = await fetchAllCourses(req.query);
+		if (response.status) {
+			return res.status(response.status).json(response);
+		}
+		const { courses, allCoursesLength } = response;
+		return res.status(200).json({ courses, allCoursesLength });
+	} catch (e) {
+		console.error('Courses fetching failed.', e);
+	}
+});
```

Comment:
Our task is not about the backend, but it's a bad idea to write routes like this.
I'll find a better article later, but for now, you can read more about node.js architecture by link(the main ideas a present in this article):
https://www.codementor.io/@evanbechtol/node-service-oriented-architecture-12vjt9zs9i

Also, you need to send the response back if an error occurred instead of "console.log". You can find a simple example in this article:
https://dev.to/qbentil/how-to-write-custom-error-handler-middleware-in-expressjs-using-javascript-29j1

Reply:
Thanks for usefull links!